core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1836,7 +1836,7 @@ class SparkContext(config: SparkConf) extends Logging {
       val uriToUse = if (!isLocal && scheme == "file") uri else new URI(key)
       val uriToDownload = UriBuilder.fromUri(uriToUse).fragment(null).build()
       val source = Utils.fetchFile(uriToDownload.toString, Utils.createTempDir(), conf,
-        hadoopConfiguration, timestamp, useCache = false, shouldUntar = false)
+        hadoopConfiguration, timestamp, useCache = true, shouldUntar = false)
```

Comment:
We don't need this archive file under root directory - the archive file itself is removed. we only need the unpacked directory

Reply:
Let me check about this.

Reply:
Seems already unpacked, remove this change